blueprints/nvidia-gpu-workload/README.md

```diff
@@ -0,0 +1,220 @@
+
+```
+terraform -chdir="../../cluster/terraform" output -raw cluster_name
+terraform -chdir="../../cluster/terraform" output -raw node_instance_role_name
+```
+
+> ***NOTE***: If you're not using Terraform, you need to get those values manually. `CLUSTER_NAME` is the name of your EKS cluster (not the ARN). Karpenter auto-generates the [instance profile](https://docs.aws.amazon.com/IAM/latest/UserGuide/id_roles_use_switch-role-ec2_instance-profiles) in your `EC2NodeClass` given the role that you specify in [spec.role](https://karpenter.sh/preview/concepts/nodeclasses/) with the placeholder `KARPENTER_NODE_IAM_ROLE_NAME`, which is a way to pass a single IAM role to the EC2 instance launched by the Karpenter `NodePool`. Typically, the instance profile name is the same as the IAM role(not the ARN).
+
+Once the placeholders are complete, to apply the EC2NodeClass and NodePool execute the following:
+```
+$> kubectl apply -f gpu-nodeclass.yaml
+ec2nodeclass.karpenter.k8s.aws/gpu created
+
+$> kubectl apply -f gpu-nodepool.yaml
+nodepool.karpenter.sh/gpu created
+```
+Now let’s deploy a test workload to see how Karpenter launches the GPU node.
+
+### Deploy a test workload to test GPU drivers are loaded
+
+The following Pod manifest launches a pod and calls the NVIDIA systems management CLI to check if a GPU is detected and the driver versions printed to standard output, which you can see when you check the logs, like this: `kubectl logs pod/nvidia-smi`.
+```
+apiVersion: v1
+kind: Pod
+metadata:
+  name: nvidia-smi
+spec:
+  nodeSelector:
+    intent: gpu
+    karpenter.k8s.aws/instance-gpu-name: "t4"
+  restartPolicy: OnFailure
+  containers:
+  - name: nvidia-smi
+    image: public.ecr.aws/amazonlinux/amazonlinux:2023-minimal
+    args:
+    - "nvidia-smi"
+    resources:
+      requests:
+        memory: "8Gi"
+        cpu: "3500m"
+      limits:
+        memory: "8Gi"
+        nvidia.com/gpu: 1
+  tolerations:
+  - key: nvidia.com/gpu
+    effect: NoSchedule
+    operator: Exists
+```
+As GPU-based workloads are likely sensitive to different GPUs (e.g. GPU memory) we've specified a `karpenter.k8s.aws/instance-gpu-name` node selector to request an instance with a specific GPU for this workload. The following nodeSelector `karpenter.k8s.aws/instance-gpu-name: "t4"` influences Karpenter node provisioning and launch the workload on a node with a [NVIDIA T4 GPU](https://aws.amazon.com/ec2/instance-types/g4/). Review the [Karpenter documentation](https://karpenter.sh/docs/reference/instance-types/) for different Amazon EC2 instances and there labels. 
+
+To deploy the workload execute the following:
+```
+$> kubectl apply -f workload.yaml
+pod/nvidia-smi created
+```
+You can check the pods status by executing:
+```
+$> kubectl get pods
```

Comment:
I'm getting this error:

"Error from server: Get "https://10.0.76.4:10250/containerLogs/default/nvidia-smi/nvidia-smi": remote error: tls: internal error"

The pod "Completed" very fast, maybe we switch this to a Job?

Reply:
my bad, something was wrong with the cluster I was using, this works